contracts/BaseStrategy.sol

```diff
@@ -566,166 +369,50 @@ abstract contract BaseStrategy {
-     *  https://github.com/Macarse/GenericKeep3rV2/blob/master/contracts/keep3r/GenericKeep3rV2.sol).
-     * @param callCostInWei The keeper's estimated gas cost to call `harvest()` (in wei).
-     * @return `true` if `harvest()` should be called, `false` otherwise.
-     */
     function harvestTrigger(uint256 callCostInWei) public virtual view returns (bool) {
         uint256 callCost = ethToWant(callCostInWei);
         StrategyParams memory params = vault.strategies(address(this));
 
-        // Should not trigger if Strategy is not activated
         if (params.activation == 0) return false;
 
-        // Should not trigger if we haven't waited long enough since previous harvest
         if (block.timestamp.sub(params.lastReport) < minReportDelay) return false;
 
-        // Should trigger if hasn't been called in a while
         if (block.timestamp.sub(params.lastReport) >= maxReportDelay) return true;
 
-        // If some amount is owed, pay it back
-        // NOTE: Since debt is based on deposits, it makes sense to guard against large
-        //       changes to the value from triggering a harvest directly through user
-        //       behavior. This should ensure reasonable resistance to manipulation
-        //       from user-initiated withdrawals as the outstanding debt fluctuates.
         uint256 outstanding = vault.debtOutstanding();
         if (outstanding > debtThreshold) return true;
 
-        // Check for profits and losses
         uint256 total = estimatedTotalAssets();
-        // Trigger if we have a loss to report
         if (total.add(debtThreshold) < params.totalDebt) return true;
 
         uint256 profit = 0;
-        if (total > params.totalDebt) profit = total.sub(params.totalDebt); // We've earned a profit!
+        if (total > params.totalDebt) profit = total.sub(params.totalDebt);
 
-        // Otherwise, only trigger if it "makes sense" economically (gas cost
-        // is <N% of value moved)
         uint256 credit = vault.creditAvailable();
         return (profitFactor.mul(callCost) < credit.add(profit));
     }
 
-    /**
-     * @notice
-     *  Harvests the Strategy, recognizing any profits or losses and adjusting
-     *  the Strategy's position.
-     *
-     *  In the rare case the Strategy is in emergency shutdown, this will exit
-     *  the Strategy's position.
-     *
-     *  This may only be called by governance, the strategist, or the keeper.
-     * @dev
-     *  When `harvest()` is called, the Strategy reports to the Vault (via
-     *  `vault.report()`), so in some cases `harvest()` must be called in order
-     *  to take in profits, to borrow newly available funds from the Vault, or
-     *  otherwise adjust its position. In other cases `harvest()` must be
-     *  called to report to the Vault on the Strategy's position, especially if
-     *  any losses have occurred.
-     */
     function harvest() external onlyKeepers {
         uint256 profit = 0;
```

Comment:
can we have an override method called, default does nothing but can be extended by strategist

`savePreHarvestState()`

and on line after 440 before ending the harvest call an override method called

`internalHealthCheck()` and default returns true and can be extended so we can use
`require(internalHealthCheck());` after PPS check

Reply:
my only concernt above is it adds size to contract which some strats are already at max, the other way i can think is if we can return a dynamic uint256 array with 10 params in memory like

`uint256[] savedState = getPreHarvestState();`
// returns empty array by default up to strategist to pack in there what values they want to check
and give meaning to each index
...
`require(internalHealthCheck(savedState));`

Not sure if that makes sense